anomaly-detection/README.MD

```diff
@@ -0,0 +1,224 @@
+```
+
+With the above command, the job will be cancelled and the applications current state will be saved in backed filesystem.
+
+2) To resume from a savepoint run the below command
+
+```
+bin/flink run -s <SAVEPOINT_LOCATION> -c io.pravega.anomalydetection.ApplicationMain <PATH_TO_pravega-flink-anomaly-detection-0.1.0-SNAPSHOT-all.jar> --configDir <APP_CONFIG_DIR> --mode 3
+```
+The job should nicely recover from the last checkpointed state. In otherwords, you should not see any alerts (duplicate) that are already processed before taking the snapshot. The alerts can be either visualized from Kibana or from Taskmanager log file.  
+
+
+## Application Configuration
+
+```
+{
+  "name": "anomaly-detection",
+  "producer": {
+    "latencyInMilliSec": 2000,
+    "capacity": 100,
+    "errorProbFactor": 0.3,
+    "controlledEnv": false
+  },
+  "pipeline": {
+    "parallelism": 1,
+    "checkpointIntervalInMilliSec": 1000,
+    "stateCheckpointDir": "hdfs://127.0.0.1:8020/checkpoint",
+    "disableCheckpoint": false,
+    "watermarkOffsetInSec": 0,
+    "windowIntervalInSeconds": 30,
+    "elasticSearch": {
+      "sinkResults": true,
+      "host": "127.0.0.1",
+      "port": 9300,
+      "cluster": "elasticsearch",
+      "index": "anomaly-index",
+      "type":"anomalies"
+    }
+  },
+  "pravega": {
+    "controllerUri": "tcp://127.0.0.1:9090",
+    "stream": "NetworkPacket",
+    "scope": "Network"
+  }
+}
+
+```
+
+`latencyInMilliSec` - how frequently events needs to be generated and published to Pravega
+
+`capacity` - initial capacity till which the error records will not be generated
+
+`errorProbFactor` - how frequently error records needs to be simulated. Provide a value between 0.1 to 1
+
+`controlledEnv` - When this is true, the `errorProbFactor` value will be ignored and the error record will be generated only when user sends request (Possible Options: `S` to simulate error record, `Q` to quit ingestion)
+
+`parallelism` - This value will be used to define pravega segment count/fixed scaling policy and Flink job parallelism
+
+`controllerUri` - Pravega controller endpoint
+ 
+`elasticSearch` - Final results will be sinked to elasticsearch if `sinkResults` is set to `true`
+ 
+`windowIntervalInSeconds` - Window frequency interval
+ 
+`watermarkOffsetInSec` - Window watermark offset interval
+ 
+`stateCheckpointDir` - HDFS FS URI along with the checkpoint directory name
+ 
+ 
+ ### Steps to setup Elastic Search/Kibana
+ 
+ 
+The steps are tested on Ubuntu 16.x OS/docker 1.13.1/JDK8/ElasticSearch5/Kibana5.3.0
+ 
+**Install Elastic Search and Kibana**
+
+```
+Install Elastic Search
+sudo mkdir -p /elk-stack/esdata
+sudo chmod 777 -R /elk-stack/
+docker run -d --name elasticsearch  --net="host" -v /elk-stack/esdata:/usr/share/elasticsearch/data elasticsearch
+
+Verify if ES is running by executing the command: 
+curl -X GET http://localhost:9200
+
+update /etc/hosts with 127.0.0.1 elasticsearch (this is for Kibana to lookup ES)
```

Comment:
There must be a better way, i.e. linked containers.   Editing the hosts file is scary to recommend, in my opinion.